shared.go

```diff
@@ -0,0 +1,147 @@
+package gojwt
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/square/go-jose/v3"
+	"github.com/square/go-jose/v3/jwt"
+)
+
+//ContextKey used to access values in the context
+type ContextKey string
+
+const (
+	//ContextJWTError is used to access the JWT error in the context
+	ContextJWTError ContextKey = "JWT_ERROR"
+	//ContextJWTObject is used to access the JWT object in the context
+	ContextJWTObject ContextKey = "JWT_OBJECT"
+	//ContextJWTClaims is used to access the JWT claims in the context
+	ContextJWTClaims ContextKey = "JWT_CLAIMS"
+	//ContextValidatorError is used to access the error provided by validators
+	ContextValidatorError ContextKey = "VALIDATOR_ERROR"
+	//ContextJWTInfo is used to get jwt info from the context
+	ContextJWTInfo ContextKey = "JWT_INFO"
+	//ContextJWTResolverError is used to get a error from the resolver
+	ContextJWTResolverError ContextKey = "JWT_RESOLVER_ERROR"
+
+	//ErrInvalidToken is returned as a error if the token was invalid
+	ErrInvalidToken = "The token provided was invalid"
+)
+
+//GetJWTInfoFromOpenIDURL will take a openid url and return a JWTinfo
+func GetJWTInfoFromOpenIDURL(OpenIDURL string) (*JWTInfo, error) {
+	wellknown, err := GetOpenIDConfigFromURL(OpenIDURL)
+	if err != nil {
+		return nil, err
+	}
+	keys, err := GetJWKSetFromOpenIDURL(wellknown.JWKSURL)
+	if err != nil {
+		return nil, err
+	}
+	jwtInfo := JWTInfo{
+		JWKS:   keys,
+		Issuer: wellknown.Issuer,
+	}
+	return &jwtInfo, nil
+}
+
+func GetOpenIDConfigFromURL(OpenIDURL string) (*OpenIDWellKnown, error) {
```

Comment:
Codacy found an issue: [exported function GetOpenIDConfigFromURL should have comment or be unexported](https://app.codacy.com/manual/johnfg10/gojwt/pullRequest?prid=5899569)